images/kindnetd/cmd/kindnetd/cni.go

```diff
@@ -34,19 +36,47 @@ import (
 type CNIConfigInputs struct {
 	PodCIDR      string
 	DefaultRoute string
+	Mtu          int
+}
+
+var mtu int
+
+func init() {
```

Comment:
init() and global data is a big anti-pattern.
mtu should be plumbed down from main.

Reply:
I didn't realize this was an anti-pattern, sorry. If you have a quick article on best practices, I'd love to read it! If not, no sweat. 


I can pull it down from main, but I think it would need to go in `makeNodesReconciler` as an extra parameter which then propagates to the `ComputeCNIConfigInputs` function. 

Alternatively we can put `mtu` as a field in `CNIConfigWriter` and in this method `func (c *CNIConfigWriter) Write(inputs CNIConfigInputs)` 

we can have `inputs.Mtu = c.mtu`

Let me know what you would prefer.

Reply:
I lean to the `CNIConfigWriter` since mtu only matter to that type.
what do you think Ben?

Reply:
no sweat :-)
I would say "avoid globals" is a good practice for all languages, I don't have a reference handy just this moment though, sorry.

`init` is best avoided in code because it's both global behavior and unpredictable import side-effects (importing the package causes this code to be run during the binary startup, in some random order versus other packages)

CNIConfigWriter makes sense to me 👍

Reply:
https://stackoverflow.com/a/56039373 is a pretty decent short reference on this point specifically 😅